cmake/sitl.cmake

```diff
@@ -69,6 +69,7 @@ if(NOT MACOSX)
     endif()
 else()
     set(SITL_COMPILE_OPTIONS ${SITL_COMPILE_OPTIONS}
+        -Wno-error=double-promotion
```

Comment:
Do we need this? Are we just missing and f after some constants? I would rather fix the warning than remove it.

Reply:
Only the MacOS compiler complains here.
Not with constants, it complains about that:
```double foo = floatVal - floatVal2;```

I (unfortunately) need double praeyision here.

Reply:
changing it to  (double)(floatVal - floatVal2) and will make the compiler happy that you are promoting it on purpose. 
The warning is probably there because there may be a performance penalty doing double math instead of float.

Reply:
The FC does not have hardware double precision, only float. This is why we have such warnings.

Reply:
Would make sense, but oddly enough the error comes from the MacOS compiler, the ARM, Linux and Windows GCC have nothing to complain about.

Reply:
The warning is not a bad thing. Even if it is only llvm on macos SITL is generating the warning I would rather have it point this out than let it go unoticed.
The fact gcc is not complaining, does not mean gcc is not promoting the operation to use double math.